From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 5 Aug 2025 00:09:31 +0200
Subject: feat(Makefile): introduce distclean target

Add a `distclean` target to remove all generated files including the one
(`Makefile.inc`) created by `configure`. Calling `make distclean` should
have the same result as `git clean -fdx`.

Forwarded: https://github.com/dracut-ng/dracut-ng/pull/1515
---
 Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1c815b4..972ac8d 100644
--- a/Makefile
+++ b/Makefile
@@ -65,7 +65,7 @@ man8pages = man/dracut.8 \
 
 manpages = $(man1pages) $(man5pages) $(man7pages) $(man8pages)
 
-.PHONY: install clean archive test all check AUTHORS CONTRIBUTORS doc
+.PHONY: install clean distclean archive test all check AUTHORS CONTRIBUTORS doc
 
 all: dracut.pc dracut-install src/skipcpio/skipcpio dracut-util
 
@@ -294,6 +294,9 @@ clean:
 	$(RM) -rf build/ doc_site/modules/ROOT/pages/man/*
 	$(MAKE) -C test clean
 
+distclean: clean
+	$(RM) Makefile.inc
+
 syncheck:
 	@ret=0;for i in dracut-initramfs-restore.sh modules.d/*/*.sh; do \
                 [ "$${i##*/}" = "module-setup.sh" ] && continue; \
